system/Database/MySQLi/Result.php

```diff
@@ -87,16 +87,52 @@ public function getFieldNames(): array
 	 */
 	public function getFieldData(): array
 	{
+		static $data_types = [
+			MYSQLI_TYPE_DECIMAL		=> 'decimal',
+			MYSQLI_TYPE_NEWDECIMAL	=> 'newdecimal',
+			MYSQLI_TYPE_FLOAT		=> 'float',
+			MYSQLI_TYPE_DOUBLE		=> 'double',
+			
+			MYSQLI_TYPE_BIT			=> 'bit',
+			MYSQLI_TYPE_TINY		=> 'tiny',
+			MYSQLI_TYPE_SHORT		=> 'short',
+			MYSQLI_TYPE_LONG		=> 'long',
+			MYSQLI_TYPE_LONGLONG	=> 'longlong',
+			MYSQLI_TYPE_INT24		=> 'int24',
+			
+			MYSQLI_TYPE_YEAR		=> 'year',
+			
+			MYSQLI_TYPE_TIMESTAMP	=> 'timestamp',
+			MYSQLI_TYPE_DATE		=> 'date',
+			MYSQLI_TYPE_TIME		=> 'time',
+			MYSQLI_TYPE_DATETIME	=> 'datetime',
+			MYSQLI_TYPE_NEWDATE		=> 'newdate',
+			
+			MYSQLI_TYPE_INTERVAL	=> 'interval',
+			MYSQLI_TYPE_SET			=> 'set',
+			MYSQLI_TYPE_ENUM		=> 'enum',
+			
+			MYSQLI_TYPE_VAR_STRING	=> 'var_string',
+			MYSQLI_TYPE_STRING		=> 'string',
+			MYSQLI_TYPE_CHAR		=> 'char',
+			
+			MYSQLI_TYPE_GEOMETRY	=> 'geometry',
+			MYSQLI_TYPE_TINY_BLOB	=> 'tiny_blob',
+			MYSQLI_TYPE_MEDIUM_BLOB	=> 'medium_blob',
+			MYSQLI_TYPE_LONG_BLOB	=> 'long_blob',
+			MYSQLI_TYPE_BLOB		=> 'blob',
+		];
 		$retVal    = [];
 		$fieldData = $this->resultID->fetch_fields();
 
 		foreach ($fieldData as $i => $data)
 		{
 			$retVal[$i]              = new \stdClass();
 			$retVal[$i]->name        = $data->name;
-			$retVal[$i]->type        = $data->type;
+			$retVal[$i]->type        = $data_types[$data->type];
```

Comment:
Please, leave the `type` as it is and add additional variable `typeName`.
```suggestion
			$retVal[$i]->type        = $data->type;
			$retVal[$i]->typeName    = $data_types[$data->type] ?? null;
```

Reply:
Make sense, will look at